packages/dashboard/src/components/DashboardProvider/IncomingRequest.tsx

```diff
@@ -104,12 +107,14 @@ function IncomingRequest({ provider, socket, request, setRequests }: Props) {
 
   const body = <div>{formatDashboardProviderRequestParameters(request)}</div>;
 
-  const footer = (
-    <div className="flex justify-start items-center gap-2">
-      <Button onClick={process} text="Process" />
-      <Button onClick={reject} text="Reject" />
-    </div>
-  );
+  const footer = disable
+    ? <div className="flex justify-start items-center gap-2">
+        <Button disabled={disable} onClick={() => {}} text="Processing..." />
```

Comment:
For properties that you're setting to true, you can just do it like this:
```suggestion
        <Button disabled onClick={() => {}} text="Processing..." />
```